internal/models/forwarder_v2_splunk.go

```diff
@@ -0,0 +1,83 @@
+package models
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+)
+
+type ForwarderSplunkV2 struct {
+	Type     string `json:"type" enum:"splunk"`
+	Endpoint string `json:"endpoint"`
+	Token    string `json:"token"`
+}
+
+func (f *ForwarderSplunkV2) call(ctx context.Context, record *LogRecord) error {
+	// Convert map[string]string to map[string]interface{}
+	eventFields := make(map[string]interface{})
+	for k, v := range record.Fields {
+		eventFields[k] = v
+	}
+
+	// Create Splunk HEC payload
+	payload := struct {
+		Event      map[string]interface{} `json:"event"`
+		Sourcetype string                 `json:"sourcetype"`
+		Source     string                 `json:"source"`
+		Host       string                 `json:"host"`
+		Time       int64                  `json:"time"`
+	}{
+		Event:      eventFields,
+		Sourcetype: "json",
+		Source:     "flowg",
+		Host:       record.Fields["host"], // or get from system
+		Time:       record.Timestamp.Unix(),
+	}
+
+	jsonData, err := json.Marshal(payload)
+	if err != nil {
+		return fmt.Errorf("failed to marshal log record: %w", err)
+	}
+
+	// Create HTTP request
+	req, err := http.NewRequestWithContext(ctx, "POST", f.Endpoint, bytes.NewBuffer(jsonData))
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+
+	// Add required headers
+	req.Header.Add("Authorization", "Splunk "+f.Token)
+	req.Header.Add("Content-Type", "application/json")
+
+	// Send request
+	client := http.Client{
+		Timeout: 5 * time.Second,
```

Comment:
You create a request with a context, you could use the default client here, and if the context has a timeout, it will automatically cancel the request.